js/source/source_cache.js

```diff
@@ -316,7 +316,8 @@ class SourceCache extends Evented {
                 minzoom: this._source.minzoom,
                 maxzoom: this._source.maxzoom,
                 roundZoom: this._source.roundZoom,
-                reparseOverscaled: this._source.reparseOverscaled
+                reparseOverscaled: this._source.reparseOverscaled,
+                wrapAroundWorld: this.style ? this.style.wrapAroundWorld : true
```

Comment:
Can `wrapAroundWorld` be a property on `transform` (the `update` method parameter), so that `Style` doesn't have to be involved?